connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1084,6 +1084,68 @@ class Dataset[T] private[sql] (
     }
   }
 
+  /**
+   * Internal helper function for building typed selects that return tuples. For simplicity and
+   * code reuse, we do this without the help of the type system and then use helper functions that
+   * cast appropriately for the user facing interface.
+   */
+  private def selectUntyped(columns: TypedColumn[_, _]*): Dataset[_] = {
+    val encoder = ProductEncoder.tuple(columns.map(_.encoder))
+    sparkSession.newDataset(encoder) { builder =>
```

Comment:
Nit you could make this is a bit DRYer if you combine some of the code with select.